src/app/pages/sharing/smb/smb-form/smb-form.component.ts

```diff
@@ -1,652 +1,399 @@
+    this.existingSmbShare = smbShare;
+    this.hostsAllowOnLoad = smbShare.hostsallow ? [...smbShare.hostsallow] : [];
+    this.mangle = smbShare.aapl_name_mangling;
+    this.hostsDenyOnLoad = smbShare.hostsdeny ? [...smbShare.hostsdeny] : [];
+    this.isTimeMachineOn = smbShare.timemachine;
+    this.title = helptextSharingSmb.formTitleEdit;
+    const index = this.namesInUse.findIndex((name) => name === smbShare.name);
+    if (index >= 0) {
+      this.namesInUse.splice(index, 1);
+    }
+    this.form.get('aapl_name_mangling').valueChanges.pipe(untilDestroyed(this)).subscribe((value) => {
+      if (value !== this.mangle && !this.mangleWarningSent) {
+        this.mangleWarningSent = true;
+        this.dialog.confirm({
+          title: helptextSharingSmb.manglingDialog.title,
+          message: helptextSharingSmb.manglingDialog.message,
+          hideCheckBox: true,
+          buttonMsg: helptextSharingSmb.manglingDialog.action,
+          hideCancel: true,
+        }).pipe(untilDestroyed(this)).subscribe();
       }
     });
+    this.form.patchValue(smbShare);
   }
 
-  afpConfirm(): void {
-    const afpControl = this.entityForm.formGroup.controls['afp'];
+  /* If user blurs name field with empty value, try to auto-populate based on path */
+  setEmptyNameFromPath(): void {
+    const nameControl = this.form.get('name');
+    if (nameControl.value) {
+      return;
+    }
+
+    const pathControl = this.form.get('path');
+    if (!pathControl.value) {
+      return;
+    }
 
+    nameControl.setValue(pathControl.value.split('/').pop());
+  }
+
+  afpConfirm(value: boolean): void {
+    if (!value) {
+      return;
+    }
+    const afpControl = this.form.get('afp');
     this.dialog.confirm({
       title: helptextSharingSmb.afpDialog_title,
       message: helptextSharingSmb.afpDialog_message,
       hideCheckBox: false,
       buttonMsg: helptextSharingSmb.afpDialog_button,
       hideCancel: false,
     }).pipe(untilDestroyed(this)).subscribe((dialogResult: boolean) => {
-      if (dialogResult) {
-        afpControl.setValue(!afpControl.value);
+      if (!dialogResult) {
+        afpControl.setValue(!value);
       }
     });
   }
 
-  showStripAclWarning(): void {
-    this.dialog.confirm({
-      title: helptextSharingSmb.stripACLDialog.title,
-      message: helptextSharingSmb.stripACLDialog.message,
-      hideCheckBox: true,
-      buttonMsg: helptextSharingSmb.stripACLDialog.button,
-      hideCancel: true,
+  submit(): void {
+    this.isLoading = true;
+    this.cdr.markForCheck();
+    const smbShare = this.form.value;
+    let request$: Observable<unknown>;
+    if (this.isNew) {
+      request$ = this.ws.call('sharing.smb.create', [smbShare]);
+    } else {
+      request$ = this.ws.call('sharing.smb.update', [this.existingSmbShare.id, smbShare]);
+    }
+
+    let cifsServiceInstance: Service;
+    request$.pipe(
+      untilDestroyed(this),
+    ).subscribe(() => {
+      this.shouldServiceRestart().pipe(
```

Comment:
Large body - hard to follow. Maybe extract some of this code into separate methods that would return observables.